variables.tf

```
@@ -186,6 +186,24 @@ variable "auto_verified_attributes" {
   ]
 }
 
+variable "account_recovery_setting_recovery_mechanisms" {
+  type        = any
+  description = "(Optional) Account recovery mechanisms"
+  # Example:
+  #
+  # account_recovery_setting_recovery_mechanisms = [
+  #   {
+  #     name          = "verified_email"
+  #     priority      = 1
+  #   },{
+  #     name          = "verified_phone_number"
+  #     priority      = 2
+  #   }]
```

I would suggest the following interface to improve UX as we can use the `length-index`  as the priority?

wdyt?

```suggestion
  # account_recovery_mechanisms = [
  #     "verified_email",
  #     "verified_phone_number",
  #   }]
```

---

I thought of that too, but having 1:1 conversions to terraform resource input is clear as well. Also a list doesn't necessarily look like it has order and needs explanation. wdyt?

---

yes, i agree, let me think a bit about this ;)

---

also not sure if they can extend the options, but if it becomes 
`{
       name         = "verified_phone_number"
       priority      = 2
       x                = y
     }
]`
, then a list(string) as datastructure will create a new challenge for backwards compatibility. Don't see a reason for AWS to extend that input map though.